Userland/Libraries/LibPDF/Value.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2021, Matthew Olsson <mattco@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Format.h>
+
+namespace PDF {
+
+class Object;
+
+class Value {
+public:
+    Value()
+        : m_type(Type::Null)
+    {
+    }
+
+    Value(bool b)
+        : m_type(Type::Bool)
+    {
+        m_as_bool = b;
+    }
+
+    Value(int i)
+        : m_type(Type::Int)
+    {
+        m_as_int = i;
+    }
+
+    Value(float f)
+        : m_type(Type::Float)
+    {
+        m_as_float = f;
+    }
+
+    template<IsObject T>
+    Value(RefPtr<T> obj)
+        : m_type(Type::Object)
+    {
+        VERIFY(obj);
```

Comment:
This feels weird. Why do we need to have a `Value(RefPtr)` constructor at all, if the `RefPtr` *must* be non-null? Seems like everyone should use the `Value(NonnullRefPtr)` one instead.